File: src/main/java/com/vspk/bookrest/exception/UserAlreadyVerifiedException.java
package com.vspk.bookrest.exception;

import lombok.Getter;

@Getter
public class UserAlreadyVerifiedException extends RuntimeException{
    private final String username;
    private final Long id;
    private final String message;

    public UserAlreadyVerifiedException(String username, String message) {
        this.username = username;
        this.message = message;
        this.id = null;
    }

    public UserAlreadyVerifiedException(Long id, String message) {
        this.id = id;
        this.message = message;
        this.username = null;
    }
}

File: src/main/java/com/vspk/bookrest/exception/UserAlreadyVerifiedException.kt
package com.vspk.bookrest.exception

class UserAlreadyVerifiedException : RuntimeException {
    val username: String?
    val id: Long?
    override val message: String

    constructor(username: String?, message: String) {
        this.username = username
        this.message = message
        id = null
    }

    constructor(id: Long?, message: String) {
        this.id = id
        this.message = message
        username = null
    }
}